packages/aws-cdk-lib/aws-autoscaling/lib/step-scaling-policy.ts

```diff
@@ -102,6 +119,18 @@ export class StepScalingPolicy extends Construct {
       throw new Error(`'scalingSteps' can have at most 40 steps, got ${props.scalingSteps.length}`);
     }
 
+    if (props.datapointsToAlarm !== undefined && !Token.isUnresolved(props.datapointsToAlarm)) {
+      if (props.datapointsToAlarm < 1) {
+        throw new Error(`datapointsToAlarm cannot be less than 1, got: ${props.datapointsToAlarm}`);
+      }
+      if (props.evaluationPeriods !== undefined
+        && !Token.isUnresolved(props.evaluationPeriods)
+        && props.evaluationPeriods < props.datapointsToAlarm
+      ) {
```

Comment:
The condition `if (props.evaluationPeriods < 1)` is not included. This is because `props.datapointsToAlarm < 1` and `props.evaluationPeriods < props.datapointsToAlarm` will cover it.

Reply:
what about if `evaluationPeriod` is set but `datapointsToAlarm` isn't?

Reply:
That's going to behave the same as it did before, so I'm not going to specifically check that.

Reply:
Did a quick check and the old behavior is a deployment failure saying `evaluationPeriod` can't be less than 1. A synth-time check would help avoid that and allow people to debug faster.

Reply:
OK, I will add the validation for `props.evaluationPeriod < 1`

Reply:
The above validations with the conditions include if `props.evaluationPeriod < 1`, so we could catch `props.evaluationPeriod < 1` in this code. Still, do we need to do anything?

Reply:
For example, evaluationPeriod for an input is 0:

- `datapoints >= 1` is guaranteed by the value pass `if (props.datapointsToAlarm < 1)` check.
- So `evaluationPeriods >= 1` is guaranteed by the value pass `evaluationPeriods < datapointsToAlarm` check and the above result (`datapoints >= 1`).

Reply:
Sorry, I misunderstood. The above conditions are in `props.datapointsToAlarm !== undefined`. So we should check regardless of whether datapointsToAlarm is undefined or not. I will add the validation!

Reply:
@aaythapa 

Changed it again and added its validation.